uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -89,12 +92,14 @@ impl Type {
             // API defined types.
             // Note that these all get unique names, and the parser ensures that the names do not
             // conflict with a builtin type. We add a prefix to the name to guard against pathological
-            // cases like a record named `SequenceRecord` interfering with `sequence<Record>`
-            Type::Object(nm) => format!("Object{}", nm),
+            // cases like a record named `SequenceRecord` interfering with `sequence<Record>`.
+            // However, types that support importing all end up with the same prefix of "Type", so
+            // that the import handling code knows how to find the remote reference.
```

Comment:
It seem worth adding more to the note here to say that it's impossible for such types to have conflicting names, which is what makes it safe for them to share the same prefix. Also I think it'd be fine to include any primitive user-defined type here even if it can't (yet) be imported, such as `Error`.